polars/polars-core/src/chunked_array/ops/repeat_by.rs

```diff
@@ -70,6 +105,17 @@ impl RepeatBy for Utf8Chunked {
 }
 impl RepeatBy for BinaryChunked {
     fn repeat_by(&self, by: &IdxCa) -> ListChunked {
+        if self.len() != by.len() {
```

Comment:
We should return an `Error`. Panics are not meant for errors.

I think we also can hoist out the error checking code in a function so we can reuse the validity checks.

Reply:
Done, but needed to change the return type for that and am not so sure if I did this correctly